Engine/SubscriptionDataReader.cs

```diff
@@ -554,23 +558,29 @@ private SubscriptionStreamReader GetReader(string source)
                     // check if this is not a local uri then download it to the local cache
                     if (uri.IsAbsoluteUri && !uri.IsLoopback)
                     {
-                        try
+                        //Skip download if the file is already in cache
+                        if (!File.Exists(location))
```

Comment:
Ok,

What do you think about adding a `bool UseLocalCache` property to `SubscriptionDataConfig` and setting it to false by default? So the above line becomes something like:

`if (!_config.UseLocalCache || !File.Exists(location))`

This way, the default behavior would be preserved and the ability to use already downloaded files (if safe) would still be there.